gobblin-compaction/src/main/java/gobblin/compaction/audit/AuditCountClientFactory.java

```diff
@@ -0,0 +1,12 @@
+package gobblin.compaction.audit;
+
+import gobblin.configuration.State;
+
+/**
+ * A factory class responsible for creating {@link AuditCountClient}
+ */
+public interface AuditCountClientFactory {
+  String AUDIT_COUNT_CLIENT_FACTORY = "audit.count.client.factory";
+  String DEFAULT_AUDIT_COUNT_CLIENT_FACTORY = "KafkaAuditCountHttpClientFactory";
```

Comment:
I don't think there should be a default. If the user doesn't specify a class, then don't run this verification.